app/message.py

```diff
@@ -59,6 +64,22 @@ def validate(cls, input_type, value):
                     f"The {input_type.lower()} value {value!r} is invalid."
                 )
 
+        # TODO validate register inputs
+
+
+class MessageTransformer:
+    @classmethod
+    def transform(cls, configuration, value):
+        tvalue = value
+        if configuration.scale:
+            value = configuration.scale * value
+            if "INT" in configuration.data_type:
```

Comment:
can you explain why is this needed?
Why do we care about int but not about floats?

Reply:
It seems like `value` is automatically a float after the multiplication operation, so if it's meant to be an int we need to turn it back into one.